include/seqan3/search/fm_index/bi_fm_index_cursor.hpp

```diff
@@ -907,9 +907,21 @@ class bi_fm_index_cursor
                       "The alphabet types of the given text and index differ.");
         assert(index != nullptr);
 
-        size_type const loc = offset() - index->fwd_fm.index[fwd_lb];
-        size_type const query_begin = loc - index->fwd_fm.text_begin_rs.rank(loc + 1) + 1; // Substract delimiters
-        return text | views::join | views::slice(query_begin, query_begin + query_length());
+        // Position of query in concatenated text.
+        size_type const location = offset() - index->fwd_fm.index[fwd_lb];
+        // The rank represents the number of delimiters before position `location + 1`.
+        // Furthermore, this number represents the i-th text this (text) position originates from, since each delimiter
+        // corresponds to one text in the whole concatenated text from left-to-right.
+        size_type const rank = index->fwd_fm.text_begin_rs.rank(location + 1);
+        // The sum of all text lengths prior to the i-th text is the start location of the i-th text in the concatenated
+        // sequence.
+        size_type const start_location = index->fwd_fm.text_begin_ss.select(rank);
+        assert(rank > 0);
+        size_type const text_id = rank - 1;
+        // Substract lengths of previous sequences.
+        size_type const query_begin = location - start_location;
+        // Take subtext, slice query out of it <-my
+        return text[text_id] | views::slice(query_begin, query_begin + query_length());
```

Comment:
Please put some spaces between the operations that belong together.

```suggestion
        // Position of query in concatenated text.
        size_type const location = offset() - index->fwd_fm.index[fwd_lb] + 1;
        // The rank represents the number of delimiters before position `location`.
        // Furthermore, this number represents the i-th text this (text) position originates from, since each delimiter
        // corresponds to one text in the whole concatenated text from left-to-right.
        size_type const rank = index->fwd_fm.text_begin_rs.rank(location);

        assert(rank > 0);

        // The sum of all text lengths prior to the i-th text is the start location of the i-th text in the concatenated
        // sequence.
        size_type const start_location = index->fwd_fm.text_begin_ss.select(rank);
        size_type const text_id = rank - 1;

        // Substract lengths of previous sequences.
        size_type const query_begin = location - start_location;

        // Take subtext, slice query out of it <-my
        return text[text_id] | views::slice(query_begin, query_begin + query_length());
```

Reply:
(Memo to myself or others: this example was wrong, because 
`size_type const location = offset() - index->fwd_fm.index[fwd_lb] + 1;` would need a `size_type const query_begin = location - 1 - start_location;` Just the spacing was left from this suggestion.)